pydantic/_internal/_validators.py

```diff
@@ -330,25 +331,90 @@ def max_length_validator(x: Any, max_length: Any) -> Any:
         raise TypeError(f"Unable to apply constraint 'max_length' to supplied value {x}")
 
 
-_CONSTRAINT_TO_VALIDATOR_LOOKUP: dict[str, Callable] = {
+def _extract_decimal_digits_info(decimal: Decimal) -> tuple[int, int]:
+    """Compute the total number of digits and decimal places for a given [`Decimal`][decimal.Decimal] instance.
+
+    This function handles both normalized and non-normalized Decimal instances.
+    Example: Decimal('1.230'), normalize=False -> 4 digits, 3 decimal places
+
+    Args:
+        decimal (Decimal): The decimal number to analyze.
+
+    Returns:
+        tuple[int, int]: A tuple containing the number of decimal places and total digits.
+
+    Modeled after the `pydanitc-core` implementation of this function.
```

Comment:
```suggestion
    Modeled after the `pydantic-core` implementation of this function.
```

though i'm not sure this is worth keeping